src/message_conversions.rs

```diff
@@ -0,0 +1,530 @@
+            get_pinned_message_base(&vsomeip_msg).set_interface_version(interface_version);
+            let payload = {
+                if let Some(bytes) = umsg.payload.clone() {
+                    bytes.to_vec()
+                } else {
+                    Vec::new()
+                }
+            };
+            let mut vsomeip_payload =
+                make_payload_wrapper(get_pinned_runtime(runtime_wrapper).create_payload());
+            set_data_safe(get_pinned_payload(&vsomeip_payload), &payload);
+            set_message_payload(&mut vsomeip_msg, &mut vsomeip_payload);
+
+            trace!(
+                "Immediately prior to request_service: service_id: {} instance_id: {}",
+                service_id,
+                instance_id
+            );
+
+            // TODO: We also need to add a corresponding stop_offer_event perhaps when we drop
+            //  the UPClientVsomeip?
+            {
+                let mut offered_events = OFFERED_EVENTS.write().await;
+                if !offered_events.contains(&(service_id, instance_id, event_id)) {
+                    get_pinned_application(application_wrapper).offer_service(
+                        service_id,
+                        instance_id,
+                        ANY_MAJOR,
+                        // interface_version,
+                        vsomeip::ANY_MINOR,
+                    );
+                    offer_single_event_safe(
+                        application_wrapper,
+                        service_id,
+                        instance_id,
+                        event_id,
+                        event_id,
+                    );
+                    tokio::time::sleep(Duration::from_nanos(1)).await;
+                    offered_events.insert((service_id, instance_id, event_id));
+                }
+            }
+
+            trace!("Immediately after request_service");
+
+            Ok(vsomeip_msg)
+        }
+        UMessageType::UMESSAGE_TYPE_REQUEST => {
+            let Some(sink) = umsg.attributes.sink.as_ref() else {
+                return Err(UStatus::fail_with_code(
+                    UCode::INVALID_ARGUMENT,
+                    "Message has no sink UUri",
+                ));
+            };
+
+            let mut vsomeip_msg =
+                make_message_wrapper(get_pinned_runtime(runtime_wrapper).create_request(true));
+            let (_instance_id, service_id) = split_u32_to_u16(sink.ue_id);
+            trace!(
+                "{} - sink.ue_id: {} source.ue_id: {} _instance_id: {} service_id:{}",
+                UP_CLIENT_VSOMEIP_FN_TAG_CONVERT_UMSG_TO_VSOMEIP_MSG,
+                sink.ue_id,
+                source.ue_id,
+                _instance_id,
+                service_id
+            );
+            get_pinned_message_base(&vsomeip_msg).set_service(service_id);
+            let instance_id = 1; // TODO: Setting to 1 manually for now
```

Comment:
IIRC from reading the SOME/IP spec, I _do_ need to respect the upper four bytes of the ue_id and assign the instance_id appropriately according to them, right? Tagging @stevenhartley & @int0x27